dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/net/UploadStatus.kt

```diff
@@ -12,13 +12,15 @@ import com.datadog.android.v2.api.InternalLogger
 internal enum class UploadStatus(val shouldRetry: Boolean) {
     SUCCESS(shouldRetry = false),
     NETWORK_ERROR(shouldRetry = true),
+    INVALID_REQUEST_ERROR(shouldRetry = false),
```

Comment:
I would suggest something like `REQUEST_CREATION_ERROR`, because `INVALID_REQUEST_ERROR` may have a meaning that request was sent, but was rejected by the server (code 400, which is currently backed by `HTTP_CLIENT_ERROR`).